app/helpers/provider/admin/dashboards_helper.rb

```diff
@@ -19,22 +19,30 @@ def sign_class(widget)
     widget.percentual_change > 0 ? 'u-plus' : 'u-minus'
   end
 
-  def dashboard_collection_link(singular_name, collection, path, plural = nil, icon_name = nil)
-    link_to path, class: 'DashboardNavigation-link' do
-      link_text = pluralize(number_to_human(collection.size), singular_name, plural)
+  def dashboard_navigation_link(link_text, path, options = {})
+    class_name = "DashboardNavigation-link #{'u-notice' if options.fetch(:notice, false)}"
+    link_to path, class: class_name do
+      icon_name = options.fetch(:icon_name, nil)
       link_text = link_text.prepend "#{icon(icon_name)} " if icon_name
       link_text.html_safe
     end
   end
 
-  def dashboard_secondary_collection_link(singular_name, collection, path, plural = nil)
+  def dashboard_secondary_link(link_text, path, options = {})
     link = ' ('
-    link << dashboard_collection_link(singular_name, collection, path, plural)
+    link << dashboard_navigation_link(link_text, path, options)
     link << ')'
   end
-
-  def dashboard_counter_link(collection, path)
-    link_to collection.size, path, class: "DashboardNavigation-link u-notice"
+  
+  def dashboard_collection_link(singular_name, collection, path, options = {})
+    link_text = pluralize(number_to_human(collection.size), singular_name, options.fetch(:plural, nil))
+    dashboard_navigation_link(link_text, path, options)
+  end
+  
+  def dashboard_secondary_collection_link(singular_name, collection, path, options = {})
+    link = ' ('
```

Comment:
Nope. This is making unsafe html fragment, that is later rendered as unescaped htm (by `==`)l. This should just return the link and the `( )` can be added in the template. 

Also, this does not return a "link" as it returns html fragment that has a link.

Reply:
If you need to add `()` around something many times, just make a helper to do that.

```ruby
def parenthesis(html)
  " (#{h html})".html_safe
end
```

Reply:
OK I can do that, do expect no less from a 3 years old piece of code ;)